dotCMS/src/main/java/com/dotcms/graphql/business/ContentAPIGraphQLFieldsProvider.java

```diff
@@ -0,0 +1,132 @@
+package com.dotcms.graphql.business;
+
+import static com.dotcms.graphql.util.TypeUtil.BASE_TYPE_SUFFIX;
+import static graphql.Scalars.GraphQLInt;
+import static graphql.Scalars.GraphQLString;
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;
+import static graphql.schema.GraphQLList.list;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.graphql.InterfaceType;
+import com.dotcms.graphql.datafetcher.ContentletDataFetcher;
+import com.dotcms.graphql.util.TypeUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.Logger;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLType;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.PredicateUtils;
+
+/**
+ * This singleton class provides all the {@link GraphQLFieldDefinition}s needed for the Content Delivery API
+ */
+enum ContentAPIGraphQLFieldsProvider implements GraphQLFieldsProvider {
+
+    INSTANCE;
+
+    @Override
+    public Collection<GraphQLFieldDefinition> getFields() throws DotDataException {
+        // Each ContentType as query'able collection field
+        List<ContentType> contentTypeList = APILocator.getContentTypeAPI(APILocator.systemUser())
+                .findAllRespectingLicense();
+
+        List<GraphQLFieldDefinition> fieldDefinitions = new ArrayList<>();
+
+        contentTypeList.forEach((type) -> {
+            fieldDefinitions.add(createCollectionField(type));
+        });
+
+        // Each BaseType as query'able collection field
+        InterfaceType.valuesAsSet().forEach((type)->{
+            fieldDefinitions.add(createCollectionFieldForBaseType(type));
+        });
+
+        CollectionUtils.filter(fieldDefinitions, PredicateUtils.notNullPredicate());
+        return fieldDefinitions;
+    }
+
+    /**
+     * Given a {@link ContentType} it creates a {@link GraphQLFieldDefinition} that represents a queryable collection for the type
+     * The name of the field will follow the convention "{typeVariable}Collection".
+     * <p>
+     * E.g Content Type Variable: Product. Resulting field name: ProductCollection
+     * @param type the type we want to build a collection for
+     * @return the field definition representing the collection
+     */
+
+    private GraphQLFieldDefinition createCollectionField(ContentType type) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'type' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)